packages/plugin-legacy/src/index.ts

```diff
@@ -165,7 +165,7 @@ function viteLegacyPlugin(options: Options = {}): Plugin[] {
     name: 'vite:legacy-config',
 
     config(config, env) {
-      if (env.command === 'build') {
+      if (env.command === 'build' && config.build?.ssr) {
```

Comment:
I think you mean this instead?
```suggestion
      if (env.command === 'build' && !config.build?.ssr) {
```

Otherwise target override happens for ssr only.

Reply:
Totally right! It was just a bad "copy&paste" from an `if (srr) return;` that was instead